mongodb/src/main/java/org/apache/metamodel/mongodb/MongoDbDataContext.java

```diff
@@ -435,6 +442,19 @@ private String getOperatorName(FilterItem item) {
         return operatorName;
     }
 
+    private Pattern turnOperandIntoRegExp(Object operand) {
+        String operandAsString = operand.toString();
+        StringBuilder operandAsRegExp = new StringBuilder(replaceWildCardLikeChars(operandAsString));
+        if (!operandAsString.startsWith("%") && operandAsString.contains("%")) {
+            operandAsRegExp = operandAsRegExp.insert(0, "^");
+        }
+        return Pattern.compile(operandAsRegExp.toString(), Pattern.CASE_INSENSITIVE);
+    }
```

Comment:
Really good point in here! I've just modified the code to always prepend ^ and append $ and the tests are all passing OK. Thank you for the comment! I'll be pushing the changes right now...